lib/index.js

```diff
@@ -79,19 +79,21 @@ internals.registrationConfig = (options) => {
     return config;
 };
 
-internals.initialize = function (server, next) {
+internals.initialize = (server, next) => {
 
-    const rootCollector = internals.state(server.root).collector;
+    const collector = internals.state(server.root).collector;
+    const rootKnex = Hoek.reach(collector, 'knexGroup.knex') || null;
 
-    rootCollector.knexGroups.forEach((knexGroup) => {
+    collector.knexGroups.forEach((knexGroup) => {
 
         const models = knexGroup.models;
+        const knex = knexGroup.knex || rootKnex;
 
         models.forEach((modelName) => {
 
-            const model = rootCollector.models[modelName];
-            const modelKnexed = model.bindKnex(knexGroup.knex);
-            rootCollector.models[modelName] = modelKnexed;
+            const Model = collector.models[modelName];
```

Comment:
I haven't completely wrapped my brain around this yet, but it feels a little weird.  What if there's a collision between `modelName`s in different `knexGroups`?  (can that even happen, or would it be caught before it got here?).

My brain's still on express' non-deterministic routing, so I'm probably just being paranoid.